ir/pass_manager.h

```diff
@@ -89,6 +89,13 @@ class PassManager : virtual public Visitor, virtual public Backtrack {
                       t();
                       return n;
                   })) {}
+
+        /// Apply the visitor held in this VisitorRef on a given IR root node. It returns the
+        /// modified node.
+        const IR::Node *process(const IR::Node *root) {
+            CHECK_NULL(visitor);
+            return root->apply(*visitor);
+        }
```

Comment:
Better to define

    Visitor *operator->() const { return visitor; }

then you can just call `->apply` (or anything else) on a VisitorRef

Maybe also define

    explicit operator bool() const { return visitor != nullptr; }